ckanext/validation/helpers.py

```diff
@@ -102,5 +102,10 @@ def get_package_id_from_resource_url():
     if match:
         return model.Package.get(match.group(1)).id
 
+def get_resource_id_from_resource_url():
+    match = re.match("/dataset/(.*)/resource/(.*)/edit", request.path)
```

Comment:
This is okay, but since it's a constant pattern, perhaps it should be precompiled at the module level with `re.compile`?